src/main/java/install/FormplayerConfigEngine.java

```diff
@@ -0,0 +1,461 @@
+import org.commcare.suite.model.*;
+import org.commcare.util.CommCarePlatform;
+import org.javarosa.core.io.BufferedInputStream;
+import org.javarosa.core.io.StreamsUtil;
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.reference.ReferenceManager;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.core.services.storage.*;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.javarosa.xpath.XPathMissingInstanceException;
+import util.PrototypeUtils;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Hashtable;
+import java.util.Vector;
+import java.util.zip.ZipFile;
+
+
+/**
+ * @author ctsims
+ *
+ */
+public class FormplayerConfigEngine {
+    private ResourceTable table;
+    private ResourceTable updateTable;
+    private ResourceTable recoveryTable;
+    private PrintStream print;
+    private CommCarePlatform platform;
+    private int fileuricount = 0;
+    private ArchiveFileRoot mArchiveRoot;
+    Log log = LogFactory.getLog(FormplayerConfigEngine.class);
+
+    public FormplayerConfigEngine(OutputStream output, final String username) {
+        this.print = new PrintStream(output);
+        this.platform = new CommCarePlatform(2, 27);
+
+        PrototypeUtils.setupPrototypes();
+
+        setRoots();
+
+        table = ResourceTable.RetrieveTable(new SqliteIndexedStorageUtility<>(Resource.class,
+                "GLOBAL_RESOURCE_TABLE", username, "dbs"));
+        updateTable = ResourceTable.RetrieveTable(new SqliteIndexedStorageUtility<>(Resource.class,
+                "UPDATE_RESOURCE_TABLE", username, "dbs"));
+        recoveryTable = ResourceTable.RetrieveTable(new SqliteIndexedStorageUtility<>(Resource.class,
+                "RECOVERY_RESOURCE_TABLE", username, "dbs"));
+
+
+        //All of the below is on account of the fact that the installers
+        //aren't going through a factory method to handle them differently
+        //per device.
+        StorageManager.forceClear();
+        StorageManager.setStorageFactory(new IStorageFactory() {
+            public IStorageUtility newStorage(String name, Class type) {
+                return new SqliteIndexedStorageUtility<>(type,
+                        name, username, "dbs");
+            }
+
+        });
+
+        StorageManager.registerStorage(Profile.STORAGE_KEY, Profile.class);
+        StorageManager.registerStorage(Suite.STORAGE_KEY, Suite.class);
+        StorageManager.registerStorage(FormDef.STORAGE_KEY,FormDef.class);
+        StorageManager.registerStorage(FormInstance.STORAGE_KEY, FormInstance.class);
+    }
+
+    private void setRoots() {
+        ReferenceManager._().addReferenceFactory(new JavaHttpRoot());
+
+        this.mArchiveRoot = new ArchiveFileRoot();
```

Comment:
i do like the `m*` variable style to denote member variables. would be nice to adopt some sort of convention